spec/system/ai_bot/persona_spec.rb

```diff
@@ -0,0 +1,50 @@
+# frozen_string_literal: true
+RSpec.describe "AI personas", type: :system, js: true do
+  fab!(:admin)
+
+  before do
+    SiteSetting.ai_bot_enabled = true
+    SiteSetting.ai_bot_enabled_chat_bots = "gpt-4|gpt-3.5-turbo"
+    sign_in(admin)
+  end
+
+  it "allows creation of a persona" do
+    visit "/admin/plugins/discourse-ai/ai_personas"
+    find(".ai-persona-list-editor__header .btn-primary").click()
+    find(".ai-persona-editor__name").set("Test Persona")
+    find(".ai-persona-editor__description").fill_in(with: "I am a test persona")
+    find(".ai-persona-editor__system_prompt").fill_in(with: "You are a helpful bot")
+    find(".ai-persona-editor__save").click()
+
+    wait_for(timeout: 2) { !page.current_path.include?("new") }
+
+    persona_id = page.current_path.split("/").last.to_i
```

Comment:
If you know you are off the new page, then you can just look up the record with the data you already have -- can just do `AiPersona.find_by(name: "Test Persona")`.

Reply:
a bit unclear on this one?

Reply:
I mean, you already know you are no longer on the New page which means the save worked, so you can just go straight to:

```ruby
persona = AiPersona.find_by(name: "Test Persona")
```

And skip having to get the ID from the URL.